modules/admediaBidAdapter.js

```diff
@@ -0,0 +1,102 @@
+import {registerBidder} from '../src/adapters/bidderFactory.js';
+import {BANNER} from '../src/mediaTypes.js';
+
+const BIDDER_CODE = 'admedia';
+const ENDPOINT_URL = 'https://prebid.admedia.com/bidder/';
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [BANNER],
+
+  /**
+   * Determines whether or not the given bid request is valid.
+   *
+   * @param {BidRequest} bid The bid params to validate.
+   * @return boolean True if this is a valid bid, and false otherwise.
+   */
+  isBidRequestValid: function (bid) {
+    return !!(bid.params.placementId);
+  },
+
+  /**
+   * Make a server request from the list of BidRequests.
+   *
+   * @return Array Info describing the request to the server.
+   * @param validBidRequests
+   * @param bidderRequest
+   */
+  buildRequests: function (validBidRequests, bidderRequest) {
+    if (validBidRequests.length === 0) {
+      return [];
+    }
+    return validBidRequests.map(bidRequest => {
+      let sizes = []
+      if (bidRequest.mediaTypes && bidRequest.mediaTypes[BANNER] && bidRequest.mediaTypes[BANNER].sizes) {
+        sizes = bidRequest.mediaTypes[BANNER].sizes;
+      }
+
+      var tagData = [];
+
+      var tagObjects = [];
+      tagObjects['tags'] = [];
+      for (var i = 0, j = sizes.length; i < j; i++) {
+        let tag = {};
+        tag.sizes = [];
+        tag.id = bidRequest.params.placementId;
+        tag.aid = bidRequest.params.aid;
+        tag.sizes.push(sizes[i].toString().replace(',', 'x'));
+        tagObjects.push(tag);
+      }
+
+      Object.assign(tagData, tagObjects);
```

Comment:
Is there any reason to copy data over from tagObjects to tagData and not just use the original here?

Reply:
Right, not needed, using original object now @mmoschovas . Thanks